assets/js/router.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * Since we've moved the router, we only want to keep routes in here
+ * that way they're all together and we don't have to scroll through
+ * multiple lines of code that was cluttering our app.js
+ */
+
+import VueRouter from "router";
+Vue.use(VueRouter);
+
+import AppComponent from "./components/app.vue";
+import BandComponent from "./components/band.vue";
+/**
+ * Mode set to history to remove the hash e.g: 
+ * /#/band/some-uuid-for-a-band -> /band/some-uuid-for-a-band
+ */
+
+
+const router = new VueRouter({
+  mode: "history",
+  routes: [
+    {path: "/band/:id", component: BandComponent, name: "showBand"},
+    {path: "/", component: AppComponent}
+  ]
+});
+
+const app = new Vue({
```

Comment:
To me, this should live in `application.js`. Especially if someone were also wiring up Vuex...but even without it, I think it makes more since to import the router into `application.js` and wire up the `app` there.